tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/DynamicConfigurationTests.cs

```diff
@@ -0,0 +1,257 @@
+// <copyright file="DynamicConfigurationTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.RemoteConfigurationManagement;
+using Datadog.Trace.Telemetry;
+using Datadog.Trace.TestHelpers;
+using Datadog.Trace.Vendors.Newtonsoft.Json;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    [CollectionDefinition(nameof(DynamicConfigurationTests), DisableParallelization = true)]
+    [Collection(nameof(DynamicConfigurationTests))]
+    public class DynamicConfigurationTests : TestHelper
+    {
+        private const string LogFileNamePrefix = "dotnet-tracer-managed-";
+        private const string DiagnosticLog = "DATADOG TRACER CONFIGURATION";
+
+        public DynamicConfigurationTests(ITestOutputHelper output)
+            : base("Console", output)
+        {
+            EnvironmentHelper.CustomEnvironmentVariables.Add(ConfigurationKeys.Telemetry.V2Enabled, "1");
+            EnvironmentHelper.CustomEnvironmentVariables.Add(ConfigurationKeys.Telemetry.HeartbeatIntervalSeconds, "1");
+        }
+
+        [SkippableFact]
+        [Trait("RunOnWindows", "True")]
+        public async Task UpdateConfiguration()
+        {
+            using var agent = EnvironmentHelper.GetMockAgent(useTelemetry: true);
+            var processName = EnvironmentHelper.IsCoreClr() ? "dotnet" : "Samples.Console";
+            using var logEntryWatcher = new LogEntryWatcher($"{LogFileNamePrefix}{processName}*");
+            using var sample = StartSample(agent, string.Empty, string.Empty, aspNetCorePort: 5000);
+
+            try
+            {
+                _ = await logEntryWatcher.WaitForLogEntry(DiagnosticLog);
+
+                await UpdateAndValidateConfig(
+                    agent,
+                    logEntryWatcher,
+                    new Config
+                    {
+                        RuntimeMetricsEnabled = true,
+                        DebugLogsEnabled = true,
+                        DataStreamsEnabled = true,
+                        LogsInjectionEnabled = true,
+                        SpanSamplingRules = "[{\"service\": \"cart*\"}]",
+                        TraceSampleRate = .5,
+                        CustomSamplingRules = "[{\"sample_rate\":0.1}]"
+                    });
```

Comment:
Should we add a test where we send the same config twice and confirm it doesn't change?

Reply:
Yeah. I know it works because the notification is processed twice (because of the RCM state issue) but it should be explicitly tested.